frontend/src/features/contacts/contact/create/Person/CreatePersonForm.tsx

```diff
@@ -0,0 +1,262 @@
+
+  // fetch autocomplete suggestions from server
+  const handleTypeaheadSearch = async (query: string) => {
+    try {
+      setIsTypeaheadLoading(true);
+      const { data } = await getOrganizationPredictions(query);
+      setMatchedOrgs(data.predictions);
+      setIsTypeaheadLoading(false);
+    } catch (e) {
+      setMatchedOrgs([]);
+      toast.error('Failed to get autocomplete results for supplied organization', {
+        autoClose: 7000,
+      });
+    } finally {
+      setIsTypeaheadLoading(false);
+    }
+  };
+
+  const onCancel = () => {
+    if (dirty) {
+      setShowConfirmation(true);
+    } else {
+      history.push('/contact/list');
+    }
+  };
+
+  return (
+    <>
+      {/* Router-based onfirmation popup when user tries to navigate away and form has unsaved changes */}
+      <UnsavedChangesPrompt />
+
+      {/* Confirmation popup when Cancel button is clicked */}
+      <GenericModal
+        title="Unsaved Changes"
+        message="Confirm cancel adding this contact? Changes will not be saved."
+        cancelButtonText="No"
+        okButtonText="Confirm"
+        display={showConfirmation}
+        handleOk={() => {
+          resetForm({ values: initialValues });
+          // need a timeout here to give the form time to reset before navigating away
+          // or else the router guard prompt will also be shown
+          setTimeout(() => history.push('/contact/list'), 100);
+        }}
+        handleCancel={() => setShowConfirmation(false)}
+      />
+
+      <Styled.CreatePersonLayout>
+        <Styled.Form id="createForm">
+          <Stack gap={1.6}>
+            <FormSection>
+              <Row>
+                <Col md={4}>
+                  <Input field="firstName" label="First Name" required />
+                </Col>
+                <Col md={3}>
+                  <Input field="middleNames" label="Middle" />
+                </Col>
+                <Col>
+                  <Input field="surname" label="Last Name" required />
+                </Col>
+              </Row>
+              <Row>
+                <Col md={7}>
+                  <Input field="preferredName" label="Preferred Name" />
+                </Col>
+                <Col></Col>
+              </Row>
+            </FormSection>
+
+            <FormSection>
+              <Styled.H2>Organization</Styled.H2>
+              <Row>
+                <Col md={7}>
+                  <AsyncTypeahead
+                    field="organizationId"
+                    label="Link to an existing organization"
+                    labelKey="text"
+                    isLoading={isTypeaheadLoading}
+                    options={matchedOrgs}
+                    onSearch={handleTypeaheadSearch}
+                  />
+                </Col>
+              </Row>
+            </FormSection>
+
+            <FormSection>
+              <Styled.H2>Contact info</Styled.H2>
+              <Styled.SummaryText
```

Comment:
seeing this I'm not in love with the idea of creating styled component wrappers like this, it seems to result in a weird mix of embedding pseudo-css in props that I don't like.

I think I'd rather we try to come up with some standardized form layouts (with Bobbi), and then codify those as css.

Of course those standardized layouts would be css so we could steal tweak them as needed, but that would give us a base at least.

Reply:
agreed

Reply:
We need base components to model basic form UI. We lack those so I attempted to create a few. This case is a component that either renders text in dark gray or red. I felt that creating two components for the color change was a waste but agree we need a set of form building blocks